stdlib/contrib/qxip/logql/logql.flux

```diff
@@ -0,0 +1,87 @@
+// Package logql provides functions meant to simplify common logql queries.
+//
+// The primary function in this package is `logql.query_range()`
+//
+// ## Metadata
+// introduced: NEXT
+//
+package logql
+
+
+import "csv"
+import "date"
+import "experimental"
+import "experimental/http/requests"
+
+// query_range queries data from a specified LogQL query within given time bounds,
+// filters data by query, timerange, and optional limit expressions.
+//
+// ## Parameters
+// - url: LogQL/qryn URL.
+// - path: LogQL query_range API path.
+// - limit: Query limit. Default is 100.
+// - query: LogQL query to execute.
+// - start: Earliest time to include in results. Default is `-1h`.
+//
+//   Results include points that match the specified start time.
+//   Use a relative duration, absolute time, or integer (Unix timestamp in nanoseconds).
+//   For example, `-1h`, `2019-08-28T22:00:00.801064Z`, or `1545136086801064`.
+//
+// - end: Latest time to include in results. Default is `now()`.
+//
+//   Results exclude points that match the specified stop time.
+//   Use a relative duration, absolute time, or integer (Unix timestamp in nanoseconds).
+//   For example, `now()`, `2019-08-28T22:00:00.801064Z`, or `1545136086801064`.
+//
```

Comment:
Related to my comment below about these two parameters. You won't be able to use unix timestamps here. You can use durations and timestamps.

```suggestion
// - start: Earliest time to include in results. Default is `-1h`.
//
//   Results include points that match the specified start time.
//   Use a relative duration or absolute time.
//   For example, `-1h` or `2022-01-01T22:00:00.801064Z`.
//
// - end: Latest time to include in results. Default is `now()`.
//
//   Results exclude points that match the specified stop time.
//   Use a relative duration or absolute time.
//   For example, `-1h` or `2022-01-01T22:00:00.801064Z`.
//
```

And I'm not sure about the behavior of these boundaries (if they're inclusive or exclusive). This is how the `start` and `stop` boundaries work in `range()`, but you may need to confirm this behavior in Loki.